src/Npgsql/Replication/ReplicationConnection.cs

```diff
@@ -700,7 +700,7 @@ async Task SendFeedback(bool waitOnSemaphore = false, bool requestReply = false,
                           $"{nameof(LastReceivedLsn)} of {new NpgsqlLogSequenceNumber(unchecked((ulong)lastReceivedLsn))}, " +
                           $"{nameof(LastFlushedLsn)} of {new NpgsqlLogSequenceNumber(unchecked((ulong)lastFlushedLsn))}, " +
                           $"{nameof(LastAppliedLsn)} of {new NpgsqlLogSequenceNumber(unchecked((ulong)lastAppliedLsn))} " +
-                          $"at {timestamp}, .", Connector.Id);
+                          $"at {timestamp.ToLocalTime()}, .", Connector.Id);
```

Comment:
I just looked at this as part of the Microsoft.Extensions.Logging work (#4247). In that PR I removed logging the timestamp altogether - as far as I can see it's just a regular timestamp at send-time, which is something that is generally handled by the user's logging library, based on their configured template. In other words, I'm not sure why we'd report the timestamp in the log message here, but not for other events; leaving it to the logger library also allows users to configure how they want their timestamps to be reported (local, UTC, whatever). Does that make sense @Brar?

https://github.com/npgsql/npgsql/pull/4247/files#diff-2dcd7e32f925efc45dec7f7a87c98d192f5d0a84d5bc574b101f3a13688fd06fR699

Reply:
> Does that make sense @Brar?

Not really.
The timestamps we log in replication methods are special because they don't only represent the time when the message was occurring but an actual value we send to the server or receive from it.
These values are meant to calculate lags between the primary and the replica and may also appear in the backend logs.
For this reason I think that we should treat this timestamp as payload and log the exact value we are sending.

Reply:
Also I don't think it makes sense to log it in [the timestamp we receive](https://github.com/npgsql/npgsql/blob/7c3b64997fc75288a90f34e8a558bbbe1721d35f/src/Npgsql/Replication/ReplicationConnection.cs#L549) and not log [the one we are sending](https://github.com/npgsql/npgsql/blob/7c3b64997fc75288a90f34e8a558bbbe1721d35f/src/Npgsql/Replication/ReplicationConnection.cs#L699).

Reply:
OK, I can see it being important to include the exact value sent to the server for e.g. comparison reasons with the PG logs. I'll add that back in #4247.

On the local vs. UTC question... Since the point is to log actual values sent and received to/from PostgreSQL, and those values seem to be UTC (I'm assuming that since we're converting them to local when receiving), shouldn't we rather be logging unmodified, UTC values in both send and receive?

Reply:
See https://github.com/npgsql/npgsql/pull/4247/commits/b3e3c8b819783dde13e732a85822a6e2eede40ed, where I added back the timestamp for sending, and also made both sides purely UTC. But if I've missed some reason they should be logged as local timestamps, I can change that.

Reply:
> On the local vs. UTC question... Since the point is to log actual values sent and received to/from PostgreSQL, and those values seem to be UTC (I'm assuming that since we're converting them to local when receiving), shouldn't we rather be logging unmodified, UTC values in both send and receive?

Good question. They are timestaptz values so I was assuming that PostgreSQL also logs them as local (based on it's own  locale setting) but I didn't check.
I think logging them as UTC is the purest way, so let's keep it that way.
We can't guarantee that the server and the client are running in the same time zone anyways.

Reply:
Makes sense, thanks @Brar.